### PR TITLE
migration: Add a case about wrong cert configuration

### PR DESCRIPTION
**Pull request by cliping**
VIRT-293985 - VM live migration - network data transport - TLS - wrong cert configuration

Signed-off-by: cliping <lcheng@redhat.com>

### CODE REVIEW

**Comment by cliping** [comment]
Test result:

#avocado run --vt-type libvirt --vt-machine-type q35 migration.migration_uri.network_data_transport.tls.wrong_cert_configuration
No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : bd12d19cc13da7396f9dd719e127e27cb69c4a51
JOB LOG    : /var/lib/avocado/job-results/job-2022-07-21T11.11-bd12d19/job.log
 (1/6) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tls.wrong_cert_configuration.no_client_cert_on_src.p2p: STARTED
 (1/6) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tls.wrong_cert_configuration.no_client_cert_on_src.p2p: PASS (175.86 s)
 (2/6) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tls.wrong_cert_configuration.no_client_cert_on_src.non_p2p: STARTED
 (2/6) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tls.wrong_cert_configuration.no_client_cert_on_src.non_p2p: PASS (175.11 s)
 (3/6) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tls.wrong_cert_configuration.no_server_cert_on_target.p2p: STARTED
 (3/6) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tls.wrong_cert_configuration.no_server_cert_on_target.p2p: PASS (177.31 s)
 (4/6) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tls.wrong_cert_configuration.no_server_cert_on_target.non_p2p: STARTED
 (4/6) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tls.wrong_cert_configuration.no_server_cert_on_target.non_p2p: PASS (178.21 s)
 (5/6) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tls.wrong_cert_configuration.wrong_hostname_in_cert.p2p: STARTED
 (5/6) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tls.wrong_cert_configuration.wrong_hostname_in_cert.p2p: PASS (176.20 s)
 (6/6) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tls.wrong_cert_configuration.wrong_hostname_in_cert.non_p2p: STARTED
 (6/6) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tls.wrong_cert_configuration.wrong_hostname_in_cert.non_p2p: PASS (175.18 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2022-07-21T11.11-bd12d19/results.html
JOB TIME   : 1061.03 s

**Comment by dzhengfy** [comment]
@cliping please fix the conflicts.

File: libvirt/tests/cfg/migration/migration_uri/migration_network_data_transport_tls_wrong_cert_configuration.cfg [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    transport_type = "tcp"
+    virsh_migrate_desturi = "qemu+tcp://${migrate_dest_host}/system"
+    custom_pki_path = "/etc/pki/qemu"
+    qemu_tls = "yes"
+    status_error = "yes"
+    virsh_migrate_extra = "--tls"
+    server_cn = "ENTER.YOUR.EXAMPLE.SERVER_CN"
+    client_cn = "ENTER.YOUR.EXAMPLE.CLIENT_CN"
+    test_case = "wrong_cert_configuration"
+
```
Comment (nanli1) [inline libvirt/tests/cfg/migration/migration_uri/migration_network_data_transport_tls_wrong_cert_configuration.cfg:36]: Could we remove this blank

File: libvirt/tests/src/migration/migration_uri/migration_network_data_transport_tls.py [hunk 2]
```diff
@@ -12,15 +15,35 @@ def run(test, params, env):
...
     vm_name = params.get("migrate_main_vm")
     vm = env.get_vm(vm_name)
     migration_obj = base_steps.MigrationBase(test, vm, params)
+    setup_test = eval("setup_%s" % test_case) if "setup_%s" % test_case in \
+        locals() else migration_obj.setup_connection
 
     try:
-        migration_obj.setup_connection()
+        setup_test()
         migration_obj.run_migration()
```
Comment (nanli1) [inline libvirt/tests/src/migration/migration_uri/migration_network_data_transport_tls.py:47]: Just wonder , where is the log sentence with test.log.info(" **** ") for run test and teardown part

**Review by dzhengfy — APPROVED** [review]
LGTM

File: libvirt/tests/cfg/migration/migration_uri/migration_network_data_transport_tls_wrong_cert_configuration.cfg (hunk 1 above)
[the remark this answers is not in the document]
Comment (cliping) [reply libvirt/tests/cfg/migration/migration_uri/migration_network_data_transport_tls_wrong_cert_configuration.cfg]: Updated.

File: libvirt/tests/src/migration/migration_uri/migration_network_data_transport_tls.py (hunk 2 above)
Comment (cliping) [reply libvirt/tests/src/migration/migration_uri/migration_network_data_transport_tls.py]: Some test.log.info() in provider/migration/base_steps.py.

**Review by nanli1 — APPROVED** [review]

**Review by dzhengfy — APPROVED** [review]
LGTM

**Pull request merged**